contentcuration/contentcuration/utils/asynccommand.py

```diff
@@ -1,16 +1,36 @@
 from abc import abstractmethod
+from collections import namedtuple
 
 from django.core.management.base import BaseCommand
+from django.core.management.base import CommandError
+
+Progress = namedtuple(
+    'Progress',
+    [
+        'progress',
+        'total',
+        'fraction',
+    ]
+)
 
 
 class TaskCommand(BaseCommand):
-    def start_progress(self, *args, **options):
-        # TODO: needs implementation
-        pass
+    def handle(self, *args, **options):
+        self.progresstracker = None
+        return self.handle_async(*args, **options)
 
-    def update_progress(self, *args, **options):
-        # TODO: needs implementation
-        pass
+    def start_progress(self, total):
+        self.progresstracker = Progress(progress=0, total=total, fraction=0)
+
+    def update_progress(self, increment):
+        tracker = self.progresstracker
+        progress = tracker.progress + increment
+        if progress > tracker.total:
+            raise CommandError("Progress reaches over 100%.")
+
+        fraction = 1.0 * progress / tracker.total
+        updated_tracker = tracker._replace(progress=progress, fraction=fraction)
```

Comment:
If we're doing a `_replace`, then there's not much use in using a `namedtuple`. I think it's better to just use a normal class at this point.